### PR TITLE
[11.0] [add] stock_warehouse_calendar

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Stock Warehouse Calendar
========================

This module adds a Calendar to the Warehouse. This calendar can then used as
the basis of the proper computation of start/end dates based on lead times in
this and other modules.

In this module, the calendar considered in the computation of start date of
stock moves and pickings created from procurements, where the lead time
is used.

Configuration
=============

* Go to *Settings* and activate the developer mode.

* Go to *Settings > Technical > Resource > Working Time* and define your
  resource calendar.

* Go to *Inventory > Configuration > Warehouse Management > Warehouses*
  and assign the Resource Calendar.

* Go to *Inventory > Configuration > Warehouse Management > Routes* and
  set up the proper delays in the procurement rules where the supply method
  is 'Move'.

Usage
=====

* When a picking is created out of a procurement evaluation (from an
  orderpoint, MTO,...) the calendar is considered in the computation of the
  expected date of the picking and moves. For example, if it takes 1 day to
  execute a stock transfer from another warehouse and it is Monday, the picking
  to resupply will be created with expected start date on the previous Friday,
  if the warehouse operates under a Mo-Fri working calendar.

### CODE REVIEW

File: stock_warehouse_calendar/models/procurement_rule.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class ProcurementRule(models.Model):
+    _inherit = 'procurement.rule'
+
+    def _get_stock_move_values(self, product_id, product_qty, product_uom,
+                               location_id, name, origin, values, group_id):
+        res = super(ProcurementRule, self)._get_stock_move_values(
+            product_id, product_qty, product_uom,
+            location_id, name, origin, values, group_id)
+
+        dt_planned = datetime.strptime(values['date_planned'],
```
Comment (pedrobaeza) [inline stock_warehouse_calendar/models/procurement_rule.py]: Use `fields.Datetime.from_string`

File: stock_warehouse_calendar/data/stock_warehouse_data.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (pedrobaeza) [inline stock_warehouse_calendar/data/stock_warehouse_data.xml]: Not needed

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Travis is red

File: stock_warehouse_calendar/README.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* Go to *Settings > Technical > Resource > Working Time* and define your
+  resource calendar.
+
+* Go to *Inventory > Configuration > Warehouse Management > Warehouses*
+  and assign the Resource Calendar.
+
+* Go to *Inventory > Configuration > Warehouse Management > Routes* and
+  set up the proper delays in the procurement rules where the supply method
+  is 'Move'.
```
Comment (MiquelRForgeFlow) [inline stock_warehouse_calendar/README.rst]:
```
set up the proper delays in the procurement rules where 'action'
is 'Move From Another Location'.
```

File: stock_warehouse_calendar/README.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+
+Configuration
+=============
+
+* Go to *Settings* and activate the developer mode.
+
+* Go to *Settings > Technical > Resource > Working Time* and define your
+  resource calendar.
+
```
Comment (MiquelRForgeFlow) [inline stock_warehouse_calendar/README.rst]: Add:

`* Go to *Inventory > Configuration > Settings* and in *Warehouse* mark 'Multi-Step Routes option'.`

File: stock_warehouse_calendar/README.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
```
Comment (MiquelRForgeFlow) [inline stock_warehouse_calendar/README.rst]: `.png`

**Review by LoisRForgeFlow — APPROVED** [review]
Tested locally and it works good! :+1:

File: stock_warehouse_calendar/data/stock_warehouse_data.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+<!-- Copyright 2018 Eficent
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+    <record id="stock.warehouse0" model="stock.warehouse">
+        <field name="calendar_id" ref="resource.resource_calendar_std"/>
```
Comment (LoisRForgeFlow) [inline stock_warehouse_calendar/data/stock_warehouse_data.xml]: maybe do not set the calendar by default?

File: stock_warehouse_calendar/__init__.py [hunk 7]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (MiquelRForgeFlow) [inline stock_warehouse_calendar/__init__.py]: this blank line could be removed

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza Could you update your review?

**Review by pedrobaeza — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
Thanks!

**Pull request merged**